modules/ding_wayf_dk/modules/ding_wayf_dk_create_user/ding_wayf_dk_create_user.module

```diff
@@ -0,0 +1,397 @@
+  // Get texts from the configuration.
+  $values = variable_get('ding_wayf_user_create_after_info', array());
+
+  return array(
+    '#theme' => 'ding_wayf_dk_info_page',
+    '#title' => empty($values['title']) ? t('Not configured.') : $values['title'],
+    '#content' => empty($values['content']) ? t('Please configure WAYF configuration to get content on this page under <a href="/admin/config/people/wayf">admin/config/people/wayf</a>.') : $values['content'],
+  );
+}
+
+/**
+ * Menu callback to create the user creation form.
+ */
+function ding_wayf_dk_create_user_form() {
+  // Remove all message from the normal login, that have not succeed.
+  drupal_get_messages(NULL, TRUE);
+
+  // Check if user have been logged in.
+  if (user_is_logged_in()) {
+    // User is logged in and have an account.
+    drupal_set_message(t('You already have an user account at the library. So we have just logged you in.'));
+
+    // Ensure session information is removed.
+    unset($_SESSION['ding_wayf_dk_create_user']);
+    unset($_SESSION['ding_wayf_dk_login_destination']);
+
+    // Send user to front page.
+    drupal_goto('<front>');
+  }
+
+  // This is a little hack to get around the panels design requirements of the
+  // ddbasic theme.
+  $form = array(
+    '#prefix' => '<div class="default-layout"><div class="layout-wrapper">',
+    '#suffix' => '<br /></div></div>',
+  );
+
+  $form['accept'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('I accept the user agreements'),
+    '#description' => variable_get('ding_wayf_user_create_accpect_text', ''),
+    '#required' => TRUE,
+  );
+
+  $form['name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Name'),
+    '#description' => t('You full name.'),
+    '#maxlength' => 100,
+    '#required' => TRUE,
+    '#attributes' => array(
+      'autocomplete' => 'off',
+    ),
+  );
+
+  $form['mail'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Email address'),
+    '#description' => t('The Email address that we can use to get in contact with you.'),
+    '#required' => TRUE,
+    '#attributes' => array(
+      'autocomplete' => 'off',
+    ),
+  );
+
+  // Get branch options form the active provider.
+  global $user;
+  $provider = _ding_provider_get_provider('reservation');
+  $options = ding_provider_invoke('reservation', 'options', $user);
+  $form['branch'] = $options[$provider['module'] . '_preferred_branch'];
+
+  $form['pin'] = array(
+    '#type' => 'password',
+    '#title' => t('Pin code'),
+    '#description' => t('Pin code with an max length of %length', array('%length' => ding_user_get_pincode_length())),
+    '#size' => 10,
+    '#maxlength' => ding_user_get_pincode_length(),
+    '#required' => TRUE,
+    '#attributes' => array(
+      'autocomplete' => 'off',
+    ),
+  );
+
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Create me'),
+  );
+
+  $link = '/wayf/logout';
+  if (user_is_logged_in()) {
+    $link = '/user/logout';
+  }
```

Comment:
Why is this necessary? How can the user be logged in to WAYF and Drupal but not have a user account?